bin/flash.flashair

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2016 Konsulko Group. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+# Make sure that the card is available to us now.
+RC=`ping -c 1 ${flashair_ip} >/dev/null 2>&1; echo $?`
+while test $RC -ne 0; do
+    RC=`ping -c 1 ${flashair_ip} >/dev/null 2>&1 ; echo $?`
+done
+
+# Make our new holding space.
+UL_DIR=`mktemp -d`
+
+# Run this script to copy files as they must appear on the target into
+# the given directory.
+. "${bin_dir}/${flashair_copy_script}"
+
+if [ ! -z "${flashair_rmlist}" ]; then
+  ARGS="rmlist:${flashair_rmlist}"
+fi
+
+ARGS="${ARGS} push:${UL_DIR}"
```

Comment:
Admittedly ${UL_DIR} is unlikely to contain spaces, but ... since everything else here should be safe if used in a directory with spaces, perhaps this could use Bash arrays rather than a single string for the args, and hence quote everything:

ARGS=()
if [ ...
    ARGS+=("xxx")
fi
ARGS+=("xxx")
push-flashair.py ${flashair_ip} "${ARGS[@]}"